ios/Classes/SwiftTruecallerSdkPlugin.swift

```diff
@@ -1,14 +1,194 @@
 import Flutter
 import UIKit
+import TrueSDK
 
-public class SwiftTruecallerSdkPlugin: NSObject, FlutterPlugin {
-  public static func register(with registrar: FlutterPluginRegistrar) {
-    let channel = FlutterMethodChannel(name: "truecaller_sdk", binaryMessenger: registrar.messenger())
-    let instance = SwiftTruecallerSdkPlugin()
-    registrar.addMethodCallDelegate(instance, channel: channel)
-  }
+private enum MethodCalls: String {
+    case initiateSDK
+    case isUsable
+    case setDarkTheme
+    case setLocale
+    case getProfile
+    case requestVerification
+    case verifyOtp
+    case verifyMissedCall
+}
+
+public class SwiftTruecallerSdkPlugin: NSObject,
+                                       FlutterPlugin {
+    private var mainChannel: FlutterMethodChannel?
+    private var eventChannel: FlutterEventChannel?
+    private var eventSink: FlutterEventSink?
+        
+    private var trueSdk = TCTrueSDK.sharedManager()
+        
+    public static func register(with registrar: FlutterPluginRegistrar) {
+        _ = SwiftTruecallerSdkPlugin(with: registrar)
+    }
+    
+    private var trueProfileResponse: TCTrueProfileResponse?
+    
+    init(with registrar: FlutterPluginRegistrar) {
+        super.init()
+        
+        addMainChannel(registrar: registrar)
+        addEventChannel(registrar: registrar)
+        registrar.addApplicationDelegate(self)
+    }
+    
+    private func addMainChannel(registrar: FlutterPluginRegistrar) {
+        mainChannel = FlutterMethodChannel(name: "tc_method_channel",
+                                           binaryMessenger: registrar.messenger())
+        guard let mainChannel = mainChannel else {
+            return
+        }
+        registrar.addMethodCallDelegate(self, channel: mainChannel)
+    }
+    
+    private func addEventChannel(registrar: FlutterPluginRegistrar) {
+        eventChannel = FlutterEventChannel(name: "tc_event_channel",
```

Comment:
name of the event channel can be moved to a constant field.

Reply:
Done